packages/plugin-builders/README.md

```diff
@@ -0,0 +1,131 @@
+
+# VMware Cloud Director Angular Plugin Builder
+
+## Overview
+The @vcd/plugin-builders package contains an Angular Builder which you can use to bundle any Angular Module in your Application in standard VMware Cloud Director UI Plugin.
+
+- [VMware Cloud Director Angular Plugin Builder](#vmware-cloud-director-angular-plugin-builder)
+	- [Overview](#overview)
+	- [Documentation](#documentation)
+		- [How it works](#how-it-works)
+		- [Angular Builder](#angular-builder)
+			- [Description](#description)
+			- [Useful References:](#useful-references)
+		- [Webpack](#webpack)
+			- [Description](#description-1)
+			- [Useful References:](#useful-references-1)
+		- [PostCSS Plugins](#postcss-plugins)
+			- [Description](#description-2)
+			- [Useful References:](#useful-references-2)
+		- [Build](#build)
+	- [Usage](#usage)
+	- [Contributing](#contributing)
+
+## Documentation
+### How it works
+Our Plugin Builder is combination of the following things:
+- Angular Builder
+- Webpack
+- PostCSS Plugins
+
+We are goin to cover each piece.
+### Angular Builder
+#### Description
+We implement our custom Angular Builder to build VMware Cloud Director UI Plugins, hence the name - Plugin Builder.
+
+We implement it in the standard Angular documented way,
+you can search for information how Custom Angular Builders are developed in the Angular documentation.
+
+#### Useful References:
+- [Angular CLI builders](https://angular.io/guide/cli-builder)
+- [Angular Builders GitHub](https://github.com/just-jeb/angular-builders)
+
+
+### Webpack
+#### Description
+We use webpack's feature `splitChunks` and `externals` to isolate the npm dependencies from your UI Plugin code, and therefore we inject in the final bundle only the source code of the UI Plugin plus custom libraries.
+
+All npm dependencis in the `externals` list will be provided to your UI Plugin by VMware Cloud Director at runtime, this boosts the loading time of your UI Plugin a lot.
+
+We leverage Webpack's `DefinePlugin`, to replace keywords in the final JS bundle. Most often we use to perform JS isolation in special type of UI Plguins called Hyper. Regular UI Plugins doesn't leverage the `DefinePlugin`.
+
+
+#### Useful References:
+- [Webpack SplitChunksPlugin](https://webpack.js.org/plugins/split-chunks-plugin/)
+- [Externals](https://webpack.js.org/configuration/externals/)
+
+### PostCSS Plugins
+#### Description
+We have 2 types of UI Plugins:
+- Regular, implemented by 3rd parties
+- Hyper, this type of UI Plugins support the regular with: JS isolation, CSS isolation and more.
+
+Now, the PostCSS Plugins are not used by the Regular UI Plugins in build time, it's used only by the Hyper ones.
```

Comment:
nit: The PostCss plugins are used at build time only by the Hyper plugins. The regular UI plugins do not utilize them.